opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -304,8 +304,10 @@ namespace Opm
                     volumeRatio += cmix_s[oilCompIdx] / b_perfcells_dense[oilCompIdx];
                 }
                 if (FluidSystem::phaseIsActive(FluidSystem::gasPhaseIdx)) {
-                    const unsigned gasCompIdx = Indices::canonicalToActiveComponentIndex(FluidSystem::gasCompIdx);
-                    volumeRatio += cmix_s[gasCompIdx] / b_perfcells_dense[gasCompIdx];
+                    if (! (FluidSystem::enableVaporizedWater() && FluidSystem::enableDissolvedGasInWater())) {
+                        const unsigned gasCompIdx = Indices::canonicalToActiveComponentIndex(FluidSystem::gasCompIdx);
```

Comment:
It is a good catch, but maybe you can put this in the else branch of this if condition above.  The logic here is rather complicated, that is why we had the fallout in the first place. 

```c++
            if (FluidSystem::enableVaporizedWater() && FluidSystem::enableDissolvedGasInWater()) {
                disOilVapWatVolumeRatio(volumeRatio, rvw, rsw, pressure,
                                        cmix_s, b_perfcells_dense, deferred_logger);
            } else {
```

Reply:
Seems like a good idea. I will try to improve the code logic.